examples/ssh/xclarity_general.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * Name: XClarity General/Health Monitoring
+ * Description: This script retrieves general and health information from Lenovo XClarity server.
+ *             
+ * Communication protocol is SSH.
+ * 
+ * Tested under Lenovo XClarity version 8.42
+ * 
+ * Creates a Custom Driver variables:
+ *      - State: The current state of the system
+ *      - Restarts: The number of system restarts or reboots
+ *      - Cooling Devices: Cooling devices and their status
+ *      - Local Storage: Local storage devices and their status
+ *      - Processors: Processors in the system
+ *      - Memory: Memory and its configuration
+ *      - System: General system information and status     
+ * 
+ */
+
+var command = "syshealth summary";
+
+// SSH options when running the command
+var sshConfig = {
+    "username": D.device.username(),
+    "password": D.device.password(),
+    "timeout": 100000,
+    "keyboard_interactive": true
+};
+
+//Handle SSH errors
+function checkSshError(err) {
+    if(err.message) console.error(err.message);
+    if(err.code == 5) D.failure(D.errorType.AUTHENTICATION_ERROR);
+    if(err.code == 255) D.failure(D.errorType.RESOURCE_UNAVAILABLE);
+    console.error(err);
+    D.failure(D.errorType.GENERIC_ERROR);
+}
+
+// Execute SSH command and return a promise
+function executeCommand(command) {
+    var d = D.q.defer();
+    sshConfig.command = command;
+    D.device.sendSSHCommand(sshConfig, function (output, err) {
+        if (err) {
+            checkSshError(err);
+        } else {
+            d.resolve(output);
+        }                  
+    });
+    return d.promise;
+}
+
+/**
+ * @remote_procedure
+ * @label Validate Association
+ * @documentation This procedure validates if the driver can be applied to a device during association and validates provided credentials.
+ */
+function validate() {
+    executeCommand(command)
+        .then(parseValidateOutput)
+        .then(D.success)
+        .catch(checkSshError);
+}
+
+function parseValidateOutput(output) {
+    if (output.trim() !== "") {
```

Comment:
This does not validate anything

eg this output will result in a successful validation

system> Error: Command not recognized. Type 'help' to get list of supported commands. system>